src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -406,19 +408,36 @@ private void updateTableSchema(TableReference tableRef) {
       FieldList sourceFields = sourceTable.getDefinition().getSchema().getFields();
       tableFieldsList = sourceFields.stream().map(Field::getName).collect(Collectors.toList());
       FieldList destinationFields = destinationTable.getDefinition().getSchema().getFields();
+      Map<String, Field> destinationFieldMap = destinationFields.stream()
+        .collect(Collectors.toMap(Field::getName, x -> x));
 
-      List<String> fieldsName = destinationFields.stream().map(Field::getName).collect(Collectors.toList());
-
-      List<Field> newFields = sourceFields.stream().filter(field -> !fieldsName.contains(field.getName()))
+      List<Field> newFields = sourceFields.stream().filter(field -> destinationFieldMap.get(field.getName()) == null)
         .collect(Collectors.toList());
+      List<Field> updateFields = new ArrayList<>();
+      sourceFields.stream()
+        .filter(sourceField -> sourceField.getMode() == null
```

Comment:
When is the mode is null? Apparently it's the same as if the mode is there and it's set to nullable? 

can you add a comment?

Reply:
Yes. For BigQuery the mode is null the same as mode set to nullable.